api.go

```diff
@@ -130,6 +131,74 @@ func (server *ApiServer) ProxyCreate(response http.ResponseWriter, request *http
 	}
 }
 
+func (server *ApiServer) Populate(response http.ResponseWriter, request *http.Request) {
+	input := []struct {
+		Proxy
+		Enabled *bool `json:"enabled"` // Overrides Proxy field to make field nullable
+	}{}
+
+	err := json.NewDecoder(request.Body).Decode(&input)
+	if apiError(response, joinError(err, ErrBadRequestBody)) {
+		return
+	}
+
+	// Check for valid input before creating any proxies
+	t := true
+	for i, p := range input {
+		if len(p.Name) < 1 {
+			apiError(response, joinError(fmt.Errorf("name"), ErrMissingField))
+			return
+		}
+		if len(p.Upstream) < 1 {
+			apiError(response, joinError(fmt.Errorf("upstream"), ErrMissingField))
+			return
+		}
+		if p.Enabled == nil {
+			input[i].Enabled = &t
+		}
+	}
+
+	proxies := make([]proxyToxics, 0, len(input))
+
+	responseCode := http.StatusCreated
+	var apiErr *ApiError
+	for _, p := range input {
+		proxy := NewProxy()
+		proxy.Name = p.Name
+		proxy.Listen = p.Listen
+		proxy.Upstream = p.Upstream
+
+		err = server.Collection.AddOrReplace(proxy, *p.Enabled)
+		if err != nil {
+			var ok bool
+			apiErr, ok = err.(*ApiError)
+			if !ok {
+				logrus.Warn("Error did not include status code: ", err)
+				apiErr = &ApiError{err.Error(), http.StatusInternalServerError}
+			}
+			responseCode = apiErr.StatusCode
+			break
+		}
+
+		proxies = append(proxies, proxyWithToxics(proxy))
```

Comment:
This is just because of how toxics are stored internally. It's really just a json wrapper so they can be displayed properly, since the internal proxy object has them in a different format / inside a collection object.